### PR TITLE
MutualGroupDms: make display consistent with Mutual Servers

### PR DESCRIPTION
**Pull request by ZoamIl**
instead of it saying "Mutual Groups" it will say "(number) Mutual Groups"

### CODE REVIEW

File: src/plugins/mutualGroupDMs/index.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ export default definePlugin({
             replacement: [
                 {
                     match: /(?<=onItemSelect:\i,children:)(\i)\.map/,
-                    replace: "[...$1, ...($self.isBotOrSelf(arguments[0].user) ? [] : [{section:'MUTUAL_GDMS',text:'Mutual Groups'}])].map"
+                    replace: "[...$1, ...($self.isBotOrSelf(arguments[0].user) ? [] : [{section:'MUTUAL_GDMS',text:`${$self.getMutualGDMCountText(arguments[0].user)}`}])].map"
```
Comment (Vendicated) [inline src/plugins/mutualGroupDMs/index.tsx:67]:
```suggestion
                    replace: "[...$1, ...($self.isBotOrSelf(arguments[0].user) ? [] : [{section:'MUTUAL_GDMS',text:$self.getMutualGDMCountText(arguments[0].user)}])].map"
```

File: src/plugins/mutualGroupDMs/index.tsx [hunk 2]
```diff
@@ -42,14 +42,14 @@ function getGroupDMName(channel: Channel) {
...
-    authors: [Devs.amia],
+    authors: [Devs.amia, Devs.Scab],
 
     patches: [
         {
             find: ".Messages.MUTUAL_GUILDS_WITH_END_COUNT", // Note: the module is lazy-loaded
             replacement: {
                 match: /(?<=\.tabBarItem.{0,50}MUTUAL_GUILDS.+?}\),)(?=.+?(\(0,\i\.jsxs?\)\(.{0,100}id:))/,
-                replace: '$self.isBotOrSelf(arguments[0].user)?null:$1"MUTUAL_GDMS",children:"Mutual Groups"}),'
+                replace: '$self.isBotOrSelf(arguments[0].user)?null:$1"MUTUAL_GDMS",children:`${$self.getMutualGDMCountText(arguments[0].user)}`}),'
```
Comment (Vendicated) [inline src/plugins/mutualGroupDMs/index.tsx:52]:
```suggestion
                replace: '$self.isBotOrSelf(arguments[0].user)?null:$1"MUTUAL_GDMS",children:$self.getMutualGDMCountText(arguments[0].user)}),'
```

File: src/plugins/mutualGroupDMs/index.tsx [hunk 3]
```diff
@@ -42,14 +42,14 @@ function getGroupDMName(channel: Channel) {
 export default definePlugin({
     name: "MutualGroupDMs",
     description: "Shows mutual group dms in profiles",
-    authors: [Devs.amia],
+    authors: [Devs.amia, Devs.Scab],
```
Comment (Vendicated) [inline src/plugins/mutualGroupDMs/index.tsx:45]: Please don't add yourself to authors for such minor changes

```suggestion
    authors: [Devs.amia],
```

Comment (ZoamIl) [reply src/plugins/mutualGroupDMs/index.tsx:45]: ok

File: src/utils/constants.ts [hunk 4]
```diff
@@ -538,6 +538,10 @@ export const Devs = /* #__PURE__*/ Object.freeze({
         name: "Joona",
         id: 297410829589020673n
     },
+    Scab: {
+        name: "Scab",
+        id: 1230764273676980317n
+    }
```
Comment (surgedevs) [inline src/utils/constants.ts]:
```suggestion
```

You forgot to also remove yourself from the devs constant after removing yourself from the plugin contributors

Comment (ZoamIl) [reply src/utils/constants.ts]: ok

File: src/plugins/mutualGroupDMs/index.tsx [hunk 5]
```diff
@@ -76,6 +76,17 @@ export default definePlugin({
...
 
+    getMutualGDMCountText: (user: User) => {
+        const count = ChannelStore.getSortedPrivateChannels().filter(c => c.isGroupDM() && c.recipients.includes(user.id)).length;
+        if (count === 0) {
+            return "No Mutual Groups";
+        } else if (count === 1) {
+            return "1 Mutual Group";
+        } else {
+            return `${count} Mutual Groups`;
+        }
```
Comment (Scyye) [inline src/plugins/mutualGroupDMs/index.tsx:87]: Why not just put "0 Mutual Groups"?
```suggestion
return `${count} Mutual Group ${count!==1?"s":""}`
```
idk how other people would feel about this minification, but :shrug:

Comment (Nuckyz) [reply src/plugins/mutualGroupDMs/index.tsx:87]: I don't remember how it is but it should match how mutual friends and servers do it

[the remark this answers is not in the document]
Comment (ZoamIl) [reply src/plugins/mutualGroupDMs/index.tsx:87]: better

[the remark this answers is not in the document]
Comment (Scyye) [reply src/plugins/mutualGroupDMs/index.tsx]: I- okay.

[the remark this answers is not in the document]
Comment (thecatontheceiling) [reply src/plugins/mutualGroupDMs/index.tsx]: this pull request is funny

[the remark this answers is not in the document]
Comment (ZoamIl) [reply src/plugins/mutualGroupDMs/index.tsx]: so funny

[the remark this answers is not in the document]
Comment (ZoamIl) [reply src/plugins/mutualGroupDMs/index.tsx]: is it even about my code anymore

**Comment by Vendicated** [comment]
thanks

**Pull request merged**